packages/@sanity/form-builder/src/sanity/DefaultAssetSource/AssetRow.tsx

```diff
@@ -0,0 +1,395 @@
+  `}
+
+  ${(props) =>
+    !props.isSelected &&
+    `
+      &:hover:before {
+        background-color: var(--card-bg-color);
+      }
+
+      &:focus:before {
+        background-color: var(--card-code-bg-color);
+      }
+
+      &:hover-within:before,
+      &:focus-within:before {
+        background-color: var(--card-bg-color);
+      }
+  `}
+`
+
+const STYLES_ROW_CARD = {position: 'relative' as any}
+const STYLES_ICON_CARD = {flexShrink: 0}
+const STYLES_BUTTON_TEXT = {minWidth: 0}
+const STYLES_ASSETMENU_WRAPPER = {zIndex: 3}
+const DISABLED_DELETE_TITLE = 'Cannot delete current file'
+
+const AssetRow = (props: RowProps) => {
+  const toast = useToast()
+  const deleteRef$ = useRef<Subscription>()
+  const [showUsageDialog, setShowUsageDialog] = useState(false)
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false)
+  const [isDeleting, setIsDeleting] = useState(false)
+  const [isOpen, setIsOpen] = useState(false)
+  const {asset, onClick, onKeyPress, onDeleteFinished, isSelected, isMobile} = props
+  const {originalFilename, _id, mimeType, size, _createdAt} = asset
+  const formattedTime = useTimeAgo(_createdAt, {agoSuffix: true})
+  const formattedMimeType = formatMimeType(mimeType)
+  const formattedSize = prettyBytes(size)
+  const showTooltip = (originalFilename || '').length > 37
+
+  const handleConfirmDelete = () => {
+    setShowDeleteDialog(true)
+  }
+
+  const handleDeleteError = (error) => {
+    toast.push({
+      closable: true,
+      status: 'error',
+      title: 'File could not be deleted',
+      description: error.message,
+    })
+  }
+
+  const handleDeleteAsset = () => {
+    setIsDeleting(true)
+
+    deleteRef$.current = versionedClient.observable.delete(asset._id).subscribe({
+      next: () => {
+        setIsDeleting(false)
+        onDeleteFinished(asset._id)
+        setShowDeleteDialog(false)
+      },
+      error: (err: Error) => {
+        setIsDeleting(false)
+        handleDeleteError(err)
+        // eslint-disable-next-line no-console
+        console.error('Could not delete asset', err)
+      },
+    })
+  }
+
+  const handleDialogClose = () => {
+    setShowUsageDialog(false)
+    setShowDeleteDialog(false)
+  }
+
+  const handleToggleUsageDialog = () => {
+    setShowUsageDialog(true)
+  }
+
+  const handleToggleOpen = () => {
+    setIsOpen(!isOpen)
+  }
+
+  const handleMenuAction = (action: AssetMenuAction) => {
+    if (action.type === 'delete') {
+      handleConfirmDelete()
+    }
+
+    if (action.type === 'showUsage') {
+      handleToggleUsageDialog()
+    }
+  }
+
+  if (isMobile) {
+    return (
+      <Card paddingBottom={2} style={STYLES_ROW_CARD}>
+        <Grid
+          columns={4}
+          gap={1}
+          style={{
+            position: 'relative',
+            gridTemplateColumns: '1fr 30px',
+            opacity: isDeleting ? 0.5 : 1,
+          }}
+        >
+          <RowButton mode="bleed" padding={0} data-id={_id} onClick={onClick} paddingY={1}>
```

Comment:
It's not currently possible to select files using the keyboard only

Reply:
This is not a regression, as this is an issue in the image select as well. Will check with the UI team to see if this is an issue upstream in the dialog or if we need to do something in particular.